### PR TITLE
Feature - add Spy logger to Testing.Logging

### LINKED ISSUE #22: Provide `SpyLogger` (stub) implementation of `ILogger`
**Is your feature request related to a problem? Please describe.**
When testing logging functionality with the `ILogger` across projects; it's sometimes useful to assert on the written messages 'in-memory' without the whole setup of a fully Serilog infrastructure.

**Describe the solution you'd like**
The `SpyLogger` exposes the written messages as a read-only sequence so the tests can assert on the expected written logs.

### PR DESCRIPTION
**Pull request by stijnmoreels**
Adds `ILogger` and `ILogger<T>` spy (stub) implementation that tracks the logged messages during testing.

Closes #22

### CODE REVIEW

File: src/Arcus.Testing.Logging/SpyLogger.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.Testing.Logging
+{
+    /// <summary>
+    /// Spy (stub) <see cref="ILogger"/> implementation to track the logged messages.
+    /// </summary>
+    public class SpyLogger : ILogger
+    {
+        private readonly ConcurrentBag<LogEntry> _entries = new ConcurrentBag<LogEntry>();
```
Comment (stijnmoreels) [inline src/Arcus.Testing.Logging/SpyLogger.cs:16]: Not sure if this should be made thread-safe, or if this is too much.

File: src/Arcus.Testing.Logging/SpyLogger.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+        /// <summary>
+        /// Gets the current logged messages.
+        /// </summary>
+        public IEnumerable<string> Messages => _entries.Select(e => e.Message).Reverse();
```
Comment (stijnmoreels) [inline src/Arcus.Testing.Logging/SpyLogger.cs:21]: Added `.Reverse()` bc of the way the bag stores things. By adding this, we can have the messages in chronocial order.

File: src/Arcus.Testing.Logging/SpyLogger.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+using System.Linq;
+using GuardNet;
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.Testing.Logging
+{
+    /// <summary>
+    /// Spy (stub) <see cref="ILogger"/> implementation to track the logged messages.
+    /// </summary>
+    public class SpyLogger : ILogger
```
Comment (tomkerkhove) [inline src/Arcus.Testing.Logging/SpyLogger.cs:14]: `Spy`, `Stub` or `InMemory`?

Comment (stijnmoreels) [reply src/Arcus.Testing.Logging/SpyLogger.cs:14]: Stub is maybe to generic for this. It's definitly used for 'spying' during testing, but it's indeed an 'in-memory' approach to do so. Maybe indeed changing to `InMemoryLogger` bc I don't know how well people are aware of the different testing terminology.

Comment (tomkerkhove) [reply src/Arcus.Testing.Logging/SpyLogger.cs:14]: Let's change it and we are good to go!

Comment (tomkerkhove) [reply src/Arcus.Testing.Logging/SpyLogger.cs:14]: If it's called SpyLogger I wouldn't find it without reading the docs for sure.

**Pull request merged**